src/noir/p2p/p2p.cpp

```diff
@@ -685,14 +685,18 @@ p2p::~p2p() {
 }
 
 void p2p::set_program_options(CLI::App& cli, CLI::App& config) {
-  config.add_option("--p2p-listen-endpoint",
+  auto p2p_options = config.add_subcommand("p2p", "Peer-to-peer configuration");
+  p2p_options->configurable();
+
+  p2p_options->add_option("p2p-listen-endpoint",
```

Comment:
You can simply bind this option to variable.

```suggestion
  p2p_options->add_option("p2p-listen-endpoint", my->p2p_address,
```